### DISCUSSION TITLE
VcDrawings画电子围栏, 编辑时候还原的问题

### DISCUSSION THREAD

**Discussion by obamin** [zouyaoji/vue-cesium#220] [General]
大佬好, 我用VcDrawings画一个面好了, 存到数据库了,  编辑的时候, 怎么还原回去给它编辑调整面呢?  我放到preRenderDatas, 除第一次加载可以, 后面再变化就无效了

**Comment by zouyaoji** [comment]
`preRenderDatas` 不是响应式的。之前也有个人有类似需求，我的解决方案是给 vc-drawings 加一个v-if条件，从数据库取数据之后要改变一下条件，应该是可行的。

**Reply by obamin** [reply]
感谢, 
v-if 方法内操作还不行, 要延迟等页面更新才行, 
使用key-changing方式, 可以马上变化了

**Reply by zouyaoji** [reply]
牛

**Reply by obamin** [reply]
> `preRenderDatas` 不是响应式的。之前也有个人有类似需求，我的解决方案是给 vc-drawings 加一个v-if条件，从数据库取数据之后要改变一下条件，应该是可行的。

那圆形是赋值什么样的初始数据呢? 
还有, 画完圆形, 我怎么拿到经纬度数据呢?

**Reply by zouyaoji** [reply]
通过 ref 模板引用能获取到哈。

vm.$refs.drawingsRef.getDrawingActionInstances()[2].cmpRef.value

**Comment by obamin** [comment]
框架已经很不错了, 要是能再完善一下围栏画图的功能就更好了. 
希望围栏完善的功能是: 
各个图形的数据可以方便的在初始加载进去,  目前看多次赋值preRenderDatas还不能刷新图形;
vc-drawings组件有方法支持拿取当前画好的图形数据, 而不是事件通知才能取数据;

见leaflet的一个画图插件操作挺好的, 要是能移植点过来就好了,
参考地址 https://github.com/geoman-io/leaflet-geoman ,  https://geoman.io/leaflet-geoman
